scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -574,25 +574,36 @@ object Decl {
 
 @branch trait Defn extends Stat
 object Defn {
+  @inline private[meta] def isPlaceholder(body: Term): Boolean =
+    body.is[Term.Placeholder]
+
   @ast class Val(
       mods: List[Mod],
       pats: List[Pat] @nonEmpty,
       decltpe: Option[sm.Type],
       rhs: Term
   ) extends Defn with Stat.WithMods with Tree.WithDeclTpeOpt with Tree.WithBody {
+    checkFields(!isPlaceholder(rhs))
     checkFields(pats.forall(!_.is[Term.Name]))
     override def body: Tree = rhs
   }
+
+  private[meta] object VarRhsCtor {
+    def apply(rhs: Option[Term]): Term = rhs.getOrElse(Term.Placeholder())
+    def toOpt(body: Term): Option[Term] = if (isPlaceholder(body)) None else Some(body)
+  }
   @ast class Var(
       mods: List[Mod],
       pats: List[Pat] @nonEmpty,
       decltpe: Option[sm.Type],
-      rhs: Option[Term]
+      @replacesFields("4.7.2", VarRhsCtor)
+      body: Term
```

Comment:
Does `var a: Int` and `var a: Int = _` become the same thing? Should we have something like an `EmptyTree` insteadto models the first case?

Reply:
`var a: Int` becomes De**cl**.Var whereas the other one is De**fn**.Var

Reply:
Ach, right! I forgot about it!